test/services/oauth2/oauth2MockServer.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const { promisify } = require('es6-promisify');
+const bodyParser = require('body-parser');
+const freeport = promisify(require('freeport'));
+
+module.exports = function oauth2MockServer({
+	port = null,
+}) {
+	const findFreePort = port ? Promise.resolve(port) : freeport();
+
+	return findFreePort.then((freePort) => new Promise((resolve) => {
+		const mockOauth = express();
+		mockOauth.use(bodyParser.json()); // for parsing application/json
+		mockOauth.use(bodyParser.urlencoded({ extended: true })); // support encoded bodies
+		mockOauth.port = freePort;
+		mockOauth.url = `http://localhost:${mockOauth.port}`;
```

Comment:
move string part to a single host constant to the beginning of the test

Reply:
@Metauriel we briefly discussed where to store the url. What is correct in this case?

Reply:
@tofixx I dont see why it would be necessary to move the constant part of the url somewhere else... the test doesnt care about it, and since this is a mock that will only ever be used for the test, it will never be anything other than localhost... 
thus I think its okay to have it static as part of the generation of the url variable, and it would be overkill to create another constant just to hold a static string value that is only used for the generation of the url...